sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -191,6 +207,27 @@ private ShouldRetryResult shouldRetryOnSessionNotAvailable() {
         }
     }
 
+    private Mono<ShouldRetryResult> shouldRetryOnStaleContainer() {
+        this.staleContainerRetryCount++;
+        if (this.rxCollectionCache == null || this.staleContainerRetryCount > 1) {
+            return Mono.just(ShouldRetryResult.noRetry());
+        }
+
+        this.request.setForceNameCacheRefresh(true);
+
+        // Refresh the sdk collection cache and throw the exception if intendedCollectionRid was passed by outside sdk, so caller will refresh their own collection cache if they have one
```

Comment:
what does " if intendedCollectionRid was passed by outside sdk" mean?

what's the scenario?

Reply:
discussed offline.